mozilla-mobile/mozilla-vpn-client
# FXVPN-32 add message (and fix bug involving lists and shared strings)

#### ~This PR cannot be merged into `main` before #10065 is merged into `main`, to ensure we don't add bad translation IDs.~ #10065 was merged.

## Description

This adds the in-app message and notification for the web extension.

Notes:
- Before this goes live, we'll need to do a few final touch ups (being tracked as FXVPN-280):
  - Confirm the pref will flip (otherwise we need to change a condition)
  - Confirm the version this will go live with (otherwise we need to change a condition)
  - Update the URL for AMO
  - Update the datestamp (if needed)
- I included some fixes for addons shared strings work, originally done as https://github.com/mozilla-mobile/mozilla-vpn-client/pull/9763. There were some bugs when using unordered or ordered lists, which have been fixed in this PR.
- I removed sample strings from the `strings.yaml` file, as we now have enough strings that we don't need those samples.
- I also fixed up some documentation

## Testing
This code will not be seen until 2.25.1. Thus, you need to change the `conditions` block. Given that QA will not be able to test this after merging to `main`, we need at least one other VPN engineer to test this code. To test:

To test:
1. Change the conditions block to the following (substituting the platform for another one, depending on what you're using):
```
  "conditions": {
    "min_client_version": "2.23.1",
    "enabled_features": ["webExtension"],
    "trigger_time": 1000,
    "platforms": ["macos"]
  },
```
2. After this is merged, we’ll get updated translation files from the translation repo. For now, manually add these lines to the end of `./3rdparty/i18n/en/addons/strings.xliff`:
```
<trans-unit id="vpn.tryFirefoxExtension.title">
        <note annotates="source" from="developer">Standard text in a composer block</note>
        <source>Try the Firefox extension</source>
      </trans-unit>
      <trans-unit id="vpn.tryFirefoxExtension.subtitle">
        <note annotates="source" from="developer">Standard text in a composer block</note>
        <source>Personalize your VPN protections to meet your web browsing needs.</source>
      </trans-unit>
      <trans-unit id="vpn.tryFirefoxExtension.bullet1">
        <note annotates="source" from="developer">Standard text in a composer block</note>
        <source>Turn off VPN for specific websites</source>
      </trans-unit>
      <trans-unit id="vpn.tryFirefoxExtension.bullet2">
        <note annotates="source" from="developer">Standard text in a composer block</note>
        <source>Set different locations for different websites, to see the web as you prefer</source>
      </trans-unit>
      <trans-unit id="vpn.tryFirefoxExtension.bullet3">
        <note annotates="source" from="developer">Standard text in a composer block</note>
        <source>Keep protection for Firefox on, even when the Mozilla VPN app is off</source>
      </trans-unit>
      <trans-unit id="vpn.tryFirefoxExtension.finalLine">
        <note annotates="source" from="developer">Standard text in a composer block</note>
        <source>Not using Firefox to browse? Give it a try. </source>
      </trans-unit>      
      <trans-unit id="vpn.tryFirefoxExtension.getExtension">
        <note annotates="source" from="developer">Standard text in a composer block</note>
        <source>Get the extension</source>
      </trans-unit>
      <trans-unit id="vpn.tryFirefoxExtension.downloadFirefox">
        <note annotates="source" from="developer">Standard text in a composer block</note>
        <source>Download Firefox</source>
      </trans-unit>
```
3. Follow the instructions in the `How to implement and test add-ons` section of `./docs/Components/Addons/index.md`.
4. Compile and run the app.
5. In the debug menu, turn on the `webExtension` feature. Hard quit and relaunch the app.
6. You should see the message and notification when you relaunch the app.

## Reference

FXVPN-32

## Checklist
    
- [x] I have not added any packages that contain high risk or unknown licenses (GPL,  LGPL, MPL, etc. consult with DevOps if in question)
- [x] I have added thorough tests where needed
